src/shared/components/palette/case-flag/write-case-flag-field.component.ts

```diff
@@ -171,20 +180,46 @@ export class WriteCaseFlagFieldComponent extends AbstractFieldWriteComponent imp
         this.errorMessages = component.errorMessages;
         this.formGroup.setErrors(component.errorMessages);
       } else {
-        // Populate new FlagDetail instance and add to the Flags data within the CaseField instance
-        if (this.fieldState === CaseFlagFieldState.FLAG_COMMENTS) {
-          const flagsCaseFieldValue = this.caseFlagParentFormGroup['caseField'].value;
-          flagsCaseFieldValue.details.push({value: this.populateNewFlagDetailInstance()});
+        switch (this.fieldState) {
+          case CaseFlagFieldState.FLAG_COMMENTS:
+            this.addFlagToCollection();
+            break;
+          case CaseFlagFieldState.FLAG_UPDATE:
+            this.updateFlagInCollection();
+            break;
+          default:
+            console.log('Invalid CaseFlagFieldState:', this.fieldState);
```

Comment:
@johnbenjamin-hmcts - I think we can remove the `console.log` statement because there shouldn't be any other states apart from these two

Reply:
Thank you @Daniel-Lam I have now updated the code.